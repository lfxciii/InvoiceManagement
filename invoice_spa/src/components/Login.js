import React, { Component } from 'react';
import { Form, Button } from 'react-bootstrap';

export default class Login extends Component {

    constructor(props) {
        super(props);

        this.state =
            {
                email: "",
                password: ""
            };
    }

    // populate state with user email and password
    handleChange = (event) => {
        this.setState({ [event.target.name]: event.target.value });
    }

    // user attempts to login
    async loginUser() {

        if (this.state.email === "" ||
            this.state.password === "") {
            alert("Please insert valid data");
        } else {

            // log user in
            let response = await fetch("auth/local", {
                method: "POST",
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({ ...this.state })
            });

            if (response.status !== 200) {
                alert("Unable to login");
            } else { // request success            
                window.location = "/loggedin";
            }
        }
    }

    // login via google
    async loginGoogle() {

        // not sure what to do here? getting cors error eventhough cors is added manually had to set mode
        let response = await fetch("http://localhost:8000/auth/google", { mode: 'no-cors' });

        if (response.status !== 200) {
            alert("Unable to redirect");
        } else { // request success                    
        }
    }

    render() {
        return (
            <Form>
                <Form.Group controlId="formBasicEmail">
                    <Form.Label>Email address</Form.Label>
                    <Form.Control type="email" onChange={this.handleChange} name="email" placeholder="Enter email" />
                    <Form.Text className="text-muted">
                        We'll never share your email with anyone else.
                    </Form.Text>
                </Form.Group>
                <Form.Group controlId="formBasicPassword">
                    <Form.Label>Password</Form.Label>
                    <Form.Control type="password" onChange={this.handleChange} name="password" placeholder="Password" />
                </Form.Group>
                <a
                    className="btn btn-success"
                    onClick={this.loginUser.bind(this)}>Login</a>

                <a
                    className="btn btn-danger m-2"
                    href="http://localhost:9000/auth/google">Google Login</a>
                <a
                    className="btn btn-primary"
                    href="http://localhost:9000/auth/google">Facebook Login</a>
            </Form>
        );
    }
}