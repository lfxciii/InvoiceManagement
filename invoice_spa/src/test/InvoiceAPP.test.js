//getCityWeatherHandler snapshot test
import React from 'react';
import Login from '../components/Login';

import renderer from 'react-test-renderer';
// var expect  = require('chai').expect;

var request = require('request');

test("test invoice render", () => {
    const tree = renderer
        .create(<Login />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('load invoices ', function(done) {
    request('http://localhost:9000/api/invoices' , function(error, response, body) {
        expect(401);
        done();
    });
});

it('load customers ', function(done) {
    request('http://localhost:9000/api/customers' , function(error, response, body) {
        expect(401);
        done();
    });
});

// test("load invoices", () => {
//     let service = new InvoiceService();

//     return service.loadInvoices()
//     .then(result => result.json())
//     .then((result) =>  {
//         expect(result.cod)
//             .toBe("200");
//     });
// });

// test("get list of customers", () => {
//     let service = new CustomerService();

//     return service.loadCustomers()
//     .then(result => result.json())
//     .then((result) =>  {
//         expect(result.cod)
//             .toBe("200");
//     });
// });