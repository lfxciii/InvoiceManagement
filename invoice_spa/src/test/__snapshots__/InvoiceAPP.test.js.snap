// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test invoice render 1`] = `
<form
  className=""
>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicEmail"
    >
      Email address
    </label>
    <input
      className="form-control"
      id="formBasicEmail"
      name="email"
      onChange={[Function]}
      placeholder="Enter email"
      type="email"
    />
    <small
      className="text-muted form-text"
    >
      We'll never share your email with anyone else.
    </small>
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicPassword"
    >
      Password
    </label>
    <input
      className="form-control"
      id="formBasicPassword"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
  </div>
  <a
    className="btn btn-success"
    onClick={[Function]}
  >
    Login
  </a>
  <a
    className="btn btn-danger m-2"
    href="http://localhost:9000/auth/google"
  >
    Google Login
  </a>
  <a
    className="btn btn-primary"
    href="http://localhost:9000/auth/google"
  >
    Facebook Login
  </a>
</form>
`;
